### PR TITLE
fix: add reason field to Details in GoogleJsonErrorObject

### PR DESCRIPTION
**Pull request by Neenu1995**
Fixes b/191502543

google-api-java-client id parsing error objects without `reason` field in `details`.
The actual error message:
```
{
  "error": {
    "code": 400,
    "message": "The subscription is in a state that is not valid for the requested operation",
    "status": "FAILED_PRECONDITION",
    "details": [
      {
        "@type": "type.googleapis.com/google.rpc.ErrorInfo",
        "reason": "ERROR_CODE_SUBSCRIPTION_BAD_STATE"
      }
    ]
  }
}
```
But when it get parsed through java api-client, it becomes:
```
{
  "code" : 400,
  "details" : [ {
    "@type" : "type.googleapis.com/google.rpc.ErrorInfo"
  } ],
  "message" : "The subscription is in a state that is not valid for the requested operation",
  "status" : "FAILED_PRECONDITION"
}
```

### CODE REVIEW

File: google-api-client-bom/pom.xml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<?xml version="1.0"?>
+<?xml version='1.0' encoding='UTF-8'?>
```
Comment (elharo) [inline google-api-client-bom/pom.xml:1]: changes in this file don't appear connected to the PR

File: google-api-client/pom.xml [hunk 2]
```diff
@@ -1,5 +1,5 @@
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+<?xml version='1.0' encoding='UTF-8'?>
```
Comment (elharo) [inline google-api-client/pom.xml:1]: revert this file

File: google-api-client/src/main/java/com/google/api/client/googleapis/json/GoogleJsonError.java [hunk 3]
```diff
@@ -207,6 +208,14 @@ public void setDetail(String detail) {
       this.detail = detail;
     }
 
+    public String getReason() {
+      return reason;
+    }
+
+    public void setReason(String reason) {
+      this.reason = reason;
```
Comment (elharo) [inline google-api-client/src/main/java/com/google/api/client/googleapis/json/GoogleJsonError.java:216]: this is supposed to be nullable?

File: pom.xml [hunk 4]
```diff
@@ -1,6 +1,5 @@
-<?xml version="1.0" encoding="UTF-8"?>
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+<?xml version='1.0' encoding='UTF-8'?>
```
Comment (elharo) [inline pom.xml:1]: revert this file, and perhaps configure your editor not to autoformat XML

File: google-api-client-bom/pom.xml (hunk 1 above)
Comment (Neenu1995) [reply google-api-client-bom/pom.xml:1]: The owlbot post processor made these chnages.

File: pom.xml (hunk 4 above)
Comment (Neenu1995) [reply pom.xml:1]: The owlbot post processor made these changes. I will revert the changes.

File: google-api-client/src/main/java/com/google/api/client/googleapis/json/GoogleJsonError.java (hunk 3 above)
Comment (Neenu1995) [reply google-api-client/src/main/java/com/google/api/client/googleapis/json/GoogleJsonError.java:216]: yes.

**Review by elharo — APPROVED** [review]

File: google-api-client/src/main/java/com/google/api/client/googleapis/json/GoogleJsonError.java (hunk 3 above)
Comment (mdsahincox) [reply google-api-client/src/main/java/com/google/api/client/googleapis/json/GoogleJsonError.java:216]: Yes

**Review by mdsahincox — APPROVED** [review]

**Review by mdsahincox — APPROVED** [review]

**Pull request merged**